apps/web-giddh/src/app/shared/advance-receipt-adjustment/advance-receipt-adjustment.component.ts

```diff
@@ -0,0 +1,515 @@
+                    taxUniqueName: '',
+                }
+            ]
+        };
+    }
+
+    /**
+     * Assign all voucher details which get from parent component
+     *
+     * @memberof AdvanceReceiptAdjustmentComponent
+     */
+    public assignVoucherDetails(): void {
+        this.adjustPayment = Object.assign(this.adjustPayment, {
+            balanceDue: Number(this.invoiceFormDetails.voucherDetails.balanceDue),
+            grandTotal: Number(this.invoiceFormDetails.voucherDetails.grandTotal),
+            customerName: this.invoiceFormDetails.voucherDetails.customerName,
+            customerUniquename: this.invoiceFormDetails.voucherDetails.customerUniquename,
+            voucherDate: this.invoiceFormDetails.voucherDetails.voucherDate,
+            totalTaxableValue: Number(this.invoiceFormDetails.voucherDetails.totalTaxableValue),
+            subTotal: Number(this.invoiceFormDetails.voucherDetails.subTotal),
+            tcsTotal: Number(this.invoiceFormDetails.voucherDetails.tcsTotal),
+            tdsTotal: Number(this.invoiceFormDetails.voucherDetails.tdsTotal)
+        });
+        if (this.getBalanceDue() > 0) {
+            this.isInvalidForm = true;
+        }
+        this.balanceDueAmount = this.invoiceFormDetails.voucherDetails.balanceDue;
+        this.offset = this.adjustPayment.balanceDue;
+    }
+
+    /**
+     * API call to get all advance receipts
+     *
+     * @memberof AdvanceReceiptAdjustmentComponent
+     */
+    public getAllAdvanceReceipts(): void {
+        if (this.adjustPayment && this.adjustPayment.customerUniquename && this.adjustPayment.voucherDate) {
+            this.getAllAdvanceReceiptsRequest.accountUniqueName = this.adjustPayment.customerUniquename;
+            this.getAllAdvanceReceiptsRequest.invoiceDate = this.adjustPayment.voucherDate;
+            this.salesService.getAllAdvanceReceiptVoucher(this.getAllAdvanceReceiptsRequest).subscribe(res => {
+                if (res.status === 'success') {
+                    this.allAdvanceReceiptResponse = res.body
+                    if (res.body && res.body.length) {
+                        this.adjustVoucherOptions = [];
+                        this.allAdvanceReceiptResponse.forEach(item => {
+                            if (item) {
+                                item.voucherDate = item.voucherDate.replace(/-/g, '/');
+                                this.adjustVoucherOptions.push({ value: item.uniqueName, label: item.voucherNumber, additional: item });
+                                this.newAdjustVoucherOptions.push({ value: item.uniqueName, label: item.voucherNumber, additional: item });
+                            }
+                        });
+                    }
+                }
+            })
+        }
+    }
+
+    /**
+     * To add new blank entry in advance receipt adjustment
+     *
+     * @memberof AdvanceReceiptAdjustmentComponent
+     */
+    public addNewBlankAdjustVoucherRow(): void {
```

Comment:
We are using return in function so we can't use void as return type.

Reply:
Done